x/rollup/keeper/deposits.go

```diff
@@ -70,15 +70,23 @@ func (k *Keeper) processL1UserDepositTxs(ctx sdk.Context, txs [][]byte) (sdk.Eve
 			return nil, types.WrapError(types.ErrInvalidL1Txs, "L1 tx must be a user deposit tx, type %d", tx.Type())
 		}
 		ctx.Logger().Debug("User deposit tx", "index", i, "tx", string(lo.Must(tx.MarshalJSON())))
-		to := tx.To()
 		// if the receipient is nil, it means the tx is creating a contract which we don't support, so return an error.
 		// see https://github.com/ethereum-optimism/op-geth/blob/v1.101301.0-rc.2/core/state_processor.go#L154
-		if to == nil {
+		if tx.To() == nil {
 			ctx.Logger().Error("Contract creation txs are not supported", "index", i)
 			return nil, types.WrapError(types.ErrInvalidL1Txs, "Contract creation txs are not supported, index:%d", i)
 		}
-		cosmAddr := utils.EvmToCosmosAddress(*to)
-		mintAmount := sdkmath.NewIntFromBigInt(tx.Value())
+
+		// Get the sender's address from the transaction
+		// We use NewLondonSigner because deposit transactions follow EIP-2718 and EIP-2930
```

Comment:
I'm not sure if it makes a difference but out of curiosity why did you pick the London signer? Should we use Cancun instead?

cc @joshklop

Reply:
We can try to use `MakeSigner`, but we need `*params.ChainConfig` for that.

Reply:
Perhaps we should move this to a global variable in the `monomer` package, until we allow more dynamic configuration on the OP Stack side.

https://github.com/polymerdao/monomer/blob/609bc9995b12777ed2d76c789bf422211bf4d586/eth/internal/ethapi/helpers.go#L37-L56

Reply:
Ah, we probably need `NewChainConfig` to be a function in the `monomer` package since we need to create a different chain config depending on the chain id.